src/target/ext/host_io.rs

```diff
@@ -0,0 +1,19 @@
+//! Provide Host I/O operations for the target.
+use crate::target::Target;
+
+/// Target Extension - Perform I/O operations on host
+pub trait HostIo: Target {
```

Comment:
We need to reconsider the error handling story for these methods.

Based on my reading of the Host I/O docs, it seems we'd want to create a new `enum HostIoError` type that maps onto https://sourceware.org/gdb/onlinedocs/gdb/Errno-Values.html#Errno-Values

I would imagine the return type to look something like:

```rust
pub enum HostIoErrno {
    EPERM = 1,
    ENOENT = 2,
    ...
}

pub enum HostIoError<E> {
    Fatal(E),
    Errno(HostIoErrno)
}

pub type HostIoResult<T, Tgt> = Result<T, HostIoError<<Tgt as Target>::Error>>;

// and have methods look like

fn open(&self, filename: &[u8], flags: u64, mode: u64) -> HostIoResult<i32, Self>;
```

That way, implementations still have the ability to return a fatal error (as they can from almost every other non-trivial `gdbstub` API), and also report a structured GDB RSP Host I/O Errno value as well.

Reply:
You can crib the Errno enum from https://github.com/luser/rust-gdb-remote-protocol/blob/master/src/fs.rs#L6-L55

Reply:
So we need a `handle_error` for `HostIoResult`?

Reply:
If you peek under the hood, you'll find that `handle_error` is just a helper method from an extension trait to streamline error handling. I'd start by manually handling `HostIoResult`, and then once you understand the kind of error handling that'll be required, you can consider creating a `host_io.rs`-local version of `handle_error` tailored for its specific needs.